diff --git a/linux/trunk/PKGBUILD b/linux/trunk/PKGBUILD
index 1895ca0c88e..fc486182337 100644
--- a/linux/trunk/PKGBUILD
+++ b/linux/trunk/PKGBUILD
@@ -19,6 +19,11 @@ source=(
   60-linux.hook  # pacman hook for depmod
   90-linux.hook  # pacman hook for initramfs regeneration
   linux.preset   # standard config files for mkinitcpio ramdisk
+  https://raw.githubusercontent.com/D0023R/linux_kernel_15khz/${_15kpatchcommitid}/linux-5.0/01_ati_9200_pllfix.diff
+  https://raw.githubusercontent.com/D0023R/linux_kernel_15khz/${_15kpatchcommitid}/linux-5.0/02_arcadevga_3000.diff
+  https://raw.githubusercontent.com/D0023R/linux_kernel_15khz/${_15kpatchcommitid}/linux-5.0/03_linux_15khz.diff
+  https://raw.githubusercontent.com/D0023R/linux_kernel_15khz/${_15kpatchcommitid}/linux-5.0/04_linux_15khz_scanoutpos.diff
+  https://raw.githubusercontent.com/D0023R/linux_kernel_15khz/${_15kpatchcommitid}/linux-5.0/05_linux_15khz_interlaced_mode_fix.diff
 )
 validpgpkeys=(
   'ABAF11C65A2970B130ABE3C479BE3E4300411886'  # Linus Torvalds
@@ -29,7 +34,12 @@ sha256sums=('SKIP'
             'fdb355884faa86a2f9725d09657c9c32573b0f1e8bb31416823d1198bcb2f80d'
             'ae2e95db94ef7176207c690224169594d49445e04249d2499e9d2fbc117a0b21'
             'c043f3033bb781e2688794a59f6d1f7ed49ef9b13eb77ff9a425df33a244a636'
-            'ad6344badc91ad0630caacde83f7f9b97276f80d26a20619a87952be65492c65')
+            'ad6344badc91ad0630caacde83f7f9b97276f80d26a20619a87952be65492c65'
+            '1cf8ddffe41e7d136767019829e11fd7732521f97f1b94f6e1e057c6f69941d8'
+            'dd09c89c7d7981b8d383d6366d98a23868e66feb9c3fc2de823a3646dda4a7a9'
+            '79a82e6653802b81514b8377565bf712295751157a43c11b0e0aa4d6b6bf4fe0'
+            '6bdf73fd06748edf9ccbef75b24c9cc078f50583509d23626c709ade2198b118'
+            '13cbd363fc82c76ef6b7a8a339d8d7018508fdb8ef22235c9324b6d4f90c7f1b')
 
 _kernelname=${pkgbase#linux}
 : ${_kernelname:=-ARCH}
@@ -46,7 +56,7 @@ prepare() {
   for src in "${source[@]}"; do
     src="${src%%::*}"
     src="${src##*/}"
-    [[ $src = *.patch ]] || continue
+    [[ $src = *.patch || $src = *.diff ]] || continue
     msg2 "Applying patch $src..."
     patch -Np1 < "../$src"
   done
